core/templates/domain/skill/skill-update.service.spec.ts

```diff
@@ -130,222 +126,312 @@ describe('Skill update service', function() {
       skill_contents: skillContentsDict,
       language_code: 'en',
       version: 3,
-      prerequisite_skill_ids: ['skill_1']
+      prerequisite_skill_ids: ['skill_1'],
     };
-  }));
-
-  it('should set/unset the skill description', function() {
-    var skill = SkillObjectFactory.createFromBackendDict(skillDict);
-    SkillUpdateService.setSkillDescription(skill, 'new description');
-    expect(UndoRedoService.getCommittableChangeList()).toEqual([{
-      cmd: 'update_skill_property',
-      property_name: 'description',
-      old_value: 'test description',
-      new_value: 'new description'
-    }]);
+  });
+
+  it('should set/unset the skill description', () => {
+    const skill = skillObjectFactory.createFromBackendDict(skillDict);
+
+    skillUpdateService.setSkillDescription(skill, 'new description');
+    expect(undoRedoService.getCommittableChangeList()).toEqual([
+      {
+        cmd: 'update_skill_property',
+        property_name: 'description',
+        old_value: 'test description',
+        new_value: 'new description',
+      },
+    ]);
+
     expect(skill.getDescription()).toEqual('new description');
-    UndoRedoService.undoChange(skill);
+    // This throws "Type 'Skill' is not assignable to parameter of type
```

Comment:
Should we pass an element of the correct type instead of bypassing this?

Reply:
Same concern

Reply:
I actually tried passing in the BackendChangeObject as expected here but it gets into the reverse block [here](https://github.com/oppia/oppia/pull/11214/files/edc54dff026f8055471c1c222cf5eca49aea9066#diff-3cd517af7176a0710b76c9e46556e397bf89ad1c9167c6b93f83ed9aec846369R159) `skill.setDescription` does not exist in BackendChangeObject. setDescription exists on Skill  not BackendChangeObject

Reply:
We tried debugging this together. There are some issues with the types in the undoRedoService that need to be fixed first. After that we can remove these ignores I think.

Reply:
Made #11370, I think that will fix the issue. Thanks!

Reply:
Okay. I have pulled your changes and removed the ts-ignore lines @nishantwrp

Reply:
Cool, thanks!